### PR TITLE
[ui-storagebrowser] updates browser url while navigating

### PR DESCRIPTION
**Pull request by ramprasadagarwal**
## What changes were proposed in this pull request?

- Encodes and updates the browser url when user navigates in file browser

## How was this patch tested?

- Added unit tests

### CODE REVIEW

**Review by JohanAhlen — APPROVED** [review]
Nice@

**Review by nidhibhatg — APPROVED** [review]
LGTM!

File: desktop/core/src/desktop/js/utils/url/changeURL.ts [hunk 1]
```diff
@@ -45,7 +43,11 @@ const changeURL = (newURL: string, params?: Record<string, string | number | boo
   } else if (window.location.hash) {
     url += window.location.hash;
   }
-  window.history.pushState(null, '', url);
+  if (isReplace) {
```
Comment (tabraiz12) [inline desktop/core/src/desktop/js/utils/url/changeURL.ts:46]: replace/push state only if new url and old url is different.

**Review by tabraiz12 — APPROVED** [review]

File: desktop/core/src/desktop/js/utils/url/changeURL.ts (hunk 1 above)
Comment (ramprasadagarwal) [reply desktop/core/src/desktop/js/utils/url/changeURL.ts:46]: Make sense, Let me add the condition.

**Pull request merged**